aten/src/ATen/native/cpu/UnaryOpsKernel.cpp

```diff
@@ -82,12 +101,21 @@ static void bitwise_not_kernel(TensorIterator& iter) {
 }
 
 static void frac_kernel(TensorIterator& iter) {
-  AT_DISPATCH_FLOATING_TYPES(iter.dtype(), "frac_cpu", [&]() {
-    cpu_kernel_vec(
-        iter,
-        [=](scalar_t a) -> scalar_t { return a - std::trunc(a); },
-        [=](Vec256<scalar_t> a) { return a.frac(); });
-  });
+  if (isComplexType(iter.dtype())) {
+    AT_DISPATCH_COMPLEX_TYPES(iter.dtype(), "frac_cpu", [&]() {
+      cpu_kernel(
+          iter,
+          [=](scalar_t a) -> scalar_t {
+            return a - scalar_t(std::trunc(std::real(a)), std::trunc(std::imag(a))); });
+    });
```

Comment:
lol, I'm not sure if this is ever actually a useful operation on complex numbers, but sure

Reply:
I agree, I will remove it. I was using it early on because it was an easy function that doesn't all `cpu_vec`.